### PR TITLE
start kubeapiserver package for sharing between kubeapiserver and federation

### PR DESCRIPTION
**Pull request by deads2k**
Removes the genericapiserver dependency on internal RBAC clients and types.

@kubernetes/sig-api-machinery-misc @sttts

### CODE REVIEW

**Comment by deads2k** [comment]
@mikedanese You're the bazel guy, right?  Any idea why bazel failed but the normal unit test passed?

```
1221 15:07:42.193] FAIL: //pkg/kubeapiserver/authorizer:go_default_test (see /root/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/execroot/kubernetes/bazel-out/local-fastbuild/testlogs/pkg/kubeapiserver/authorizer/go_default_test/test.log)
W1221 15:07:42.193] ____From Testing //pkg/kubeapiserver/authorizer:go_default_test:
I1221 15:07:42.194] ==================== Test output for //pkg/kubeapiserver/authorizer:go_default_test:
I1221 15:07:42.194] --- FAIL: TestNew (0.00s)
I1221 15:07:42.195] 	config_test.go:97: NewAuthorizerFromAuthorizationConfig errored while using a valid policy file: open ../../auth/authorizer/abac/example_policy_file.jsonl: no such file or directory
I1221 15:07:42.195] FAIL
```

**Comment by deads2k** [comment]
> @mikedanese You're the bazel guy, right? Any idea why bazel failed but the normal unit test passed?

testing a theory with the second commit.  Will squash if it works.

**Comment by deads2k** [comment]
> testing a theory with the second commit. Will squash if it works.

It didn't.  I'll need some help with it I think.

**Comment by deads2k** [comment]
> It didn't. I'll need some help with it I think.

got it sorted.  had to manually add a stanza

File: pkg/kubeapiserver/authorizer/config_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			},
+			wantErr: true,
+			msg:     "should have errored when Webhook config file is used without ModeWebhook",
+		},
+	}
+
+	for _, tt := range tests {
+		_, err := tt.config.New()
+		if tt.wantErr && (err == nil) {
+			t.Errorf("NewAuthorizerFromAuthorizationConfig %s", tt.msg)
```
Comment (sttts) [inline pkg/kubeapiserver/authorizer/config_test.go]: It's `Config.New()` now, `NewAuthorizerFromAuthorizationConfig` is gone.

**Comment by sttts** [comment]
One nit, otherwise lgtm

**Comment by deads2k** [comment]
removing tag since this is included in another pull also in the queue.

**Comment by mikedanese** [comment]
Sorry, just saw this. You need to declare the .jsonl as testdata so it's
included in the test sandbox. I'll send a PR.

**Comment by mikedanese** [comment]
ahh looks like you figured it out.

**Pull request merged**